tests/resource/test_resource.py

```diff
@@ -29,14 +30,14 @@ class TestResources:
     resource_quantities = (
         Resources(),
         Resources(5, 0, {}, 0, 0),
-        Resources(1, 3, {"Hadamard": 1, "PauliZ": 2}, 3, 10),
+        Resources(1, 3, defaultdict(int, {"Hadamard": 1, "PauliZ": 2}), 3, 10),
```

Comment:
Why this is changed? The users need to use defaultdict when they create a `Resource` object?

Reply:
Context: 
While initializing an instance of `Resources` the user can pass any dictionary for the `gate_types` argument. It's  more efficient to use a `defaultdict` in `_count_resources()` when counting the `gate_types`. Due to this, specs will return a `Resources` instance which has specifically a `defaultdict` as the type for the Resources object. 

Answer: 
Added a test to make sure this prints well when using defaultdict and the `__str__` method.